packages/xod-client/src/editor/actions.js

```diff
@@ -172,7 +201,7 @@ export const linkPin = (nodeId, pinKey) => (dispatch, getState) => {
     dispatch(addError(LINK_ERRORS[error]));
     dispatch(clearPinSelection());
   } else {
-    dispatch(addLink(linkingFrom, linkingTo));
+    dispatch(addLink(linkingFrom, linkingTo, patchPath));
```

Comment:
Rather than dealing with the current patch far above before it is required, I think fold it right where it is referred makes code more readable:

```js
dispatch(
  foldMaybe(
    clearPinSelection(),
    patchPath => addLink(linkingFrom, linkingTo, patchPath),
    Selectors.getCurrentPatchPath(state)
  )
)
```